src/Rdap.php

```diff
@@ -66,9 +121,30 @@ public function dns(): RdapDns
     {
         return $this->rdapDns;
     }
+    public function ipv4(): RdapIpV4
+    {
+        return $this->rdapIpV4;
+    }
+    public function ipv6(): RdapIpV6
+    {
+        return $this->rdapIpV6;
+    }
+    
 
     public function supportedTlds(): array
     {
         return $this->dns()->supportedTlds();
     }
+    private function getIpAndVersion(string $ip): ?array
+    {
+        $ipV4 = filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4);
+        if (!$ipV4) {
```

Comment:
Simplify this by not using nesting. Maybe test for ipv6 first, so you can directly return if it's positive?